rir/src/runtime/FunctionSignature.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "R/r.h"
+
+namespace rir {
+
+struct FunctionSignature {
```

Comment:
Or change all to classes?

Reply:
@o-, what do you think? Structs or classes?

If we change all to classes, let's do that in a separate PR.

Reply:
I don't care too much. I would keep the things which are gnur-object mapped as structs. and since the function sigunature will eventually be that way too, I think it's fine.

Reply:
at some point I was under the impression that structs are guaranteed to not have vtables, but it seems that this is actually not true.

Reply:
OK, let's use structs, and then explicitly make things private.

Reply:
I don't like private anyway :)

Reply:
I don't feel strongly enough to argue this point, and we've spent enough time here, so to keep things simple, just ignore this comment thread and leave it alone. :)